conan/api/model.py

```diff
@@ -81,26 +83,31 @@ def prefs(self):
         return prefs
 
     def add_prefs(self, prefs, configurations=None):
-        confs = configurations or OrderedDict()
         for pref in prefs:
-            binary_info = confs.get(pref, OrderedDict())
+            binary_info = configurations.get(pref) if configurations is not None else None
             self.recipes.setdefault(pref.ref, []).append((pref, binary_info))
 
-    @property
-    def ordered_recipes_by_name(self):
-        ret = OrderedDict()
-        for ref, prefs in self.recipes.items():
-            ret.setdefault(ref.name, OrderedDict()).setdefault(ref, prefs)
-        return ret
-
     def serialize(self):
-        ret = OrderedDict()
-        for ref, prefs in self.recipes.items():
-            pref_ret = OrderedDict()
-            if prefs:
-                for pref, binary_info in prefs:
-                    pref_ret[pref.repr_notime()] = binary_info
-            ret[ref.repr_notime()] = pref_ret
+        ret = {}
+        for ref, prefs in sorted(self.recipes.items()):
+            ref_dict = ret.setdefault(str(ref), {})
+            if ref.revision:
+                revisions_dict = ref_dict.setdefault("revisions", {})
+                rev_dict = revisions_dict.setdefault(ref.revision, {})
+                if ref.timestamp:
+                    rev_dict["timestamp"] = timestamp_to_str(ref.timestamp)
+                if prefs:
+                    packages_dict = rev_dict.setdefault("packages", {})
+                    for pref_info in prefs:
+                        pref, info = pref_info
+                        pid_dict = packages_dict.setdefault(pref.package_id, {})
+                        if info is not None:
+                            pid_dict["info"] = info
```

Comment:
Why could `info` not be an empty `dict` by default?

I saw this line above:

```python
binary_info = configurations.get(pref) if configurations is not None else None
```
So, I don't know if I'm missing anything else about any caveats.

Reply:
That's what I wanted to know. Thanks for the clarification! 😉